CMakeLists.txt

```diff
@@ -160,13 +166,35 @@ if(NOT CMAKE_VERSION VERSION_LESS 3.13)
   target_link_options(CLI11_warnings INTERFACE $<$<BOOL:${CLI11_FORCE_LIBCXX}>:-stdlib=libc++>)
 endif()
 
+# To see in IDE, headers must be listed for target
+set(MAYBE_CONFIGURE_DEPENDS "")
+if(CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME AND NOT CMAKE_VERSION VERSION_LESS 3.12)
+  list(INSERT MAYBE_CONFIGURE_DEPENDS 0 CONFIGURE_DEPENDS)
+endif()
+
+file(GLOB CLI11_headers ${MAYBE_CONFIGURE_DEPENDS} "${PROJECT_SOURCE_DIR}/include/CLI/*.hpp")
+file(GLOB CLI11_impl_headers ${MAYBE_CONFIGURE_DEPENDS}
+     "${PROJECT_SOURCE_DIR}/include/CLI/impl/*.hpp")
+
+if(CLI11_PRECOMPILED)
+  # Create static lib
+  file(GLOB CLI11_precompile_sources "${PROJECT_SOURCE_DIR}/src/*.cpp")
```

Comment:
Do you really think that globbing source files is a good idea?

Reply:
I have no opinion on globbing. Master branch is using glob so this PR is following the status-quo. I'm happy to update it if glob is removed from the main branch.

Reply:
Yeah - it's already bad in master but this makes it even worse when the library will get build with outdated source list during development it's time wasted for chasing non-existent problems.
I know some people do not want to list all the files in the build scripts but I'd urge the maintainers to consider that.

Reply:
`CONFIGURE_DEPENDS` makes it quite a bit better, and we add that if CMake supports it.

Reply:
(though I think it's missing here, actually)